mirgecom/boundary.py

```diff
@@ -35,16 +36,72 @@
 import numpy as np
 from meshmode.dof_array import thaw
 from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
-# from mirgecom.eos import IdealSingleGas
 from grudge.symbolic.primitives import TracePair
 from mirgecom.fluid import split_conserved, join_conserved
+from mirgecom.inviscid import inviscid_facial_flux
 
 
-class PrescribedBoundary:
-    """Boundary condition prescribes boundary soln with user-specified function.
+class FluidBoundary:
+    r"""Abstract interface to fluid boundary treatment.
+
+    .. automethod:: inviscid_boundary_flux
+    """
+
+    def inviscid_boundary_flux(self, discr, btag, q, eos, **kwargs):
+        """Get the inviscid flux across the boundary faces."""
+        raise NotImplementedError()
```

Comment:
If we want to use abstract interfaces (which I'm all for!) Then I'd suggest using [metaclasses via abc](https://docs.python.org/3/library/abc.html) So this would become:
```suggestion
from abc import ABCMeta, abstractmethod


class FluidBoundary(metaclass=ABCMeta):
    r"""Abstract interface to fluid boundary treatment.

    .. automethod:: inviscid_boundary_flux
    """

    @abstractmethod
    def inviscid_boundary_flux(self, discr, btag, q, eos, **kwargs):
        """Get the inviscid flux across the boundary faces."""
```

The advantage of this approach is that, when adding addition derived classes, objects will fail to instantiate if they do not implement the necessary abstract methods/properties. You're more or less establishing a contract will all derived classes.

Reply:
Yes, this is great.  Thanks!